chan.go

```diff
@@ -83,26 +109,29 @@ func (r *chanResponse) Next() (interface{}, error) {
 	}
 
 	select {
+	case <-r.closed:
+		return nil, r.err
 	case v, ok := <-r.ch:
 		if !ok {
+			r.ch = nil
+			if r.err != nil {
+				return nil, r.err
+			}
+
 			return nil, io.EOF
 		}
 
-		if err, ok := v.(cmdkit.Error); ok {
+		if err, ok := v.(*cmdkit.Error); ok {
```

Comment:
I'm pretty sure we just want to remove this entirely.

Reply:
I'm not sure, [here](https://github.com/ipfs/go-ipfs-cmds/pull/112#discussion_r192347895) whyrusleeping argued we should check both all the time. There's [an issue](https://github.com/ipfs/go-ipfs-cmdkit/issues/19) on fixing this for `cmdkit.Error` in general, but there was little feedback. 

Also, that line is wrong, should be without the dereference. I'll fix that.

Reply:
We're already checking if something is an error *before* putting it *into* the channel. If we leave this here, we'll need to leve the `cmdkit.Error` case in the switch statement below. However, IMO, we should just remove both (there shouldn't be *any* errors here).